addons/l10n_sa/wizard/account_payment_register.py

```diff
@@ -0,0 +1,29 @@
+from odoo import api, fields, models
+
+
+class AccountPaymentRegister(models.TransientModel):
+    _inherit = 'account.payment.register'
+
+    l10n_sa_display_outstanding_account = fields.Boolean(compute='_compute_l10n_sa_display_outstanding_account')
+
+    @api.depends('partner_id', 'payment_type', 'journal_id')
+    def _compute_l10n_sa_display_outstanding_account(self):
+        for record in self:
+            moves = record.batches[0]['lines'].move_id
+            record.l10n_sa_display_outstanding_account = record.company_id.country_code == 'SA' and not \
+                                                         record.withholding_payment_account_id and \
+                                                         any(tax.tax_exigibility == 'on_payment' for tax in moves.line_ids.tax_ids)
```

Comment:
You need to add some comments when you write these kind of things.
Why do you need yet another boolean in order to compute the display?

AFAIR `withholding_outstanding_account_id` will show in the view when it is expected to already, without changing anything.